### PR TITLE
make CUE inner kube package as a standalone package && clean code

### PR DESCRIPTION
**Pull request by wonderflow**
1. make inner kube package in a style of `kube/apps/v1`, so all the apiVersion/Kind will work, including CRDs.
2. refactor the code, use a `PackageDiscover` object to replace the global var to store the inner packages.
3. add tests and fix the CI
4. make `vela system dry-run` work on it.

### CODE REVIEW

File: pkg/dsl/model/instance.go [hunk 1]
```diff
@@ -1,6 +1,12 @@
 package model
 
 import (
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"cuelang.org/go/cue/build"
+
```
Comment (ryanzhang-oss) [inline pkg/dsl/model/instance.go:9]: goimport

I thought the IDE does this automatically

File: references/cli/init.go [hunk 2]
```diff
@@ -16,6 +16,7 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	"github.com/oam-dev/kubevela/apis/types"
+	common2 "github.com/oam-dev/kubevela/pkg/utils/common"
```
Comment (ryanzhang-oss) [inline references/cli/init.go:19]: util/comutil ?

**Review by ryanzhang-oss — APPROVED** [review]
I know it's tempting but It will be better to leave the move/rename part as a separate PR.

File: pkg/appfile/parser.go [hunk 3]
```diff
@@ -156,27 +161,30 @@ func (p *Parser) GenerateAppFile(ctx context.Context, name string, app *v1alpha2
...
 	settings, err := util.RawExtension2Map(&comp.Settings)
 	if err != nil {
 		return nil, errors.WithMessagef(err, "fail to parse settings for %s", comp.Name)
 	}
-	workload.Params = settings
+	workload := &Workload{
+		Traits:              []*Trait{},
+		Name:                comp.Name,
+		Type:                comp.WorkloadType,
+		CapabilityCategory:  templ.CapabilityCategory,
```
[the remark this answers is not in the document]
Comment (wonderflow) [reply pkg/appfile/parser.go:178]: next PR

File: pkg/dsl/model/instance.go (hunk 1 above)
Comment (wonderflow) [reply pkg/dsl/model/instance.go:9]: Do you know how to configure the IDE?

**Comment by wonderflow** [comment]
refer to https://github.com/oam-dev/kubevela/issues/1195

**Pull request merged**